tests/psp22wrapper.tests.ts

```diff
@@ -0,0 +1,151 @@
+        // sender has 900 tokens
+        await expect(uQuery.balanceOf(sender.address)).to.have.output(900)
+        // sender has 100 wrapper tokens
+        await expect(wQuery.balanceOf(sender.address)).to.have.output(100)
+        // contract has 100 tokens
+        await expect(uQuery.balanceOf(wrapper.address)).to.have.output(100)
+        
+        // sender withdraws 100 tokens
+        await expect(fromSigner(wrapper, sender.address).tx.withdrawTo(sender.address, 100)).to.eventually.be.fulfilled
+        
+        // sender has 1000 tokens
+        await expect(uQuery.balanceOf(sender.address)).to.have.output(1000)
+        // sender has 0 wrapped tokens
+        await expect(wQuery.balanceOf(sender.address)).to.have.output(0)
+        // contract has 0 tokens
+        await expect(uQuery.balanceOf(wrapper.address)).to.have.output(0)
+    })
+    
+    it('Withdraw without deposit should fail', async () => {
+        const contracts = await setup()
+        const { contract: wrapper, query: wQuery } = contracts['wrapper']
+        const { query: uQuery, defaultSigner: sender } = contracts['psp22']
+        
+        // sender has 1000 tokens and 0 wrapper tokens, contract has 0 tokens
+        await expect(uQuery.balanceOf(sender.address)).to.have.output(1000)
+        await expect(wQuery.balanceOf(sender.address)).to.have.output(0)
+        await expect(uQuery.balanceOf(wrapper.address)).to.have.output(0)
+        
+        // sender withdraws 100 tokens
+        await expect(fromSigner(wrapper, sender.address).tx.withdrawTo(sender.address, 100)).to.eventually.be.rejected
+        
+        // sender has 1000 tokens and 0 wrapper tokens, contract has 0 tokens
+        await expect(uQuery.balanceOf(sender.address)).to.have.output(1000)
+        await expect(wQuery.balanceOf(sender.address)).to.have.output(0)
+        await expect(uQuery.balanceOf(wrapper.address)).to.have.output(0)
+    })
+
+    it('Recover works', async () => {
+        const contracts = await setup()
+        const { contract: wrapper, query: wQuery } = contracts['wrapper']
+        const { contract: underlying, query: uQuery, defaultSigner: sender } = contracts['psp22']
+
+        // sender allows the contract to spend their tokens
+        await expect(underlying.tx.approve(wrapper.address, 100)).to.eventually.be.fulfilled
+        // sender wraps 100 tokens
+        await expect(fromSigner(wrapper, sender.address).tx.depositFor(sender.address, 100)).to.eventually.be.fulfilled
+        // sender has 900 tokens
+        await expect(uQuery.balanceOf(sender.address)).to.have.output(900)
+        // sender has 100 wrapper tokens
+        await expect(wQuery.balanceOf(sender.address)).to.have.output(100)
+        // contract has 100 tokens
+        await expect(uQuery.balanceOf(wrapper.address)).to.have.output(100)
+
+        // sender accidentaly burns 100 tokens
+        await expect(fromSigner(wrapper, sender.address).tx.burn(100)).to.eventually.be.fulfilled
```

Comment:
You don't need to burn it, you can transfer tokens to the account of the wrapper. Without calling of `deposit` function. It will emulate what you want